File: 1231-replace-elements-with-greatest-element-on-right-side/1231-replace-elements-with-greatest-element-on-right-side.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int result[] = new int[arr.length];
        result[arr.length-1] = -1;
        int max = -1;

        for(int i=arr.length-2; i>=0; i--){
            max = Math.max(max, arr[i+1]);
            result[i] = max;
        }

        return result;
    }
}

File: 1231-replace-elements-with-greatest-element-on-right-side/1231-replace-elements-with-greatest-element-on-right-side.py
class Solution(object):
    def replaceElements(self, arr):
        """
        :type arr: List[int]
        :rtype: List[int]
        """
        n = len(arr)
        result = [0] * n
        result[n-1] = -1
        max_val = -1

        i = n - 2
        while i >= 0:
            max_val = max(max_val, arr[i+1])
            result[i] = max_val
            i -= 1

        return result
